Dev m2k to master

nunojsa opened the pull request:
Merged ad-adc driver into master. This driver had some conflicts between m2k and master, so  that, m2k device needs to have some special handling in this driver. Also fixed some coding style warnings.
This PR ends the series of PRs to merge m2k into master.

commodo commented on drivers/iio/adc/ad_adc.c:
```diff
@@ -26,7 +26,7 @@
 struct adc_chip_info {
 	const char *name;
-	int (*special_probe) (struct platform_device *pdev);
+	int (*special_probe)(struct platform_device *pdev);
```
this was added in a the previous patch
why not fix it there ?

nunojsa replied:
Honestly because i really only realized it after running checkpatch and then just fixed it on the run.

commodo commented on drivers/iio/adc/ad_adc.c:
```diff
@@ -476,7 +529,9 @@ static int adc_probe(struct platform_device *pdev)
 	if (IS_ERR(st->clk)) {
-		return PTR_ERR(st->clk);
+		/* m2k is an exception... */
+		if (strcmp(info->name, "m2k_adc"))
```
is the name only needed for this check ?
if so, maybe let's just add a bool to `info->has_no_sampl_clk` or something;
otherwise, we would need to handle device names for new devices that don't need sample clocks

nunojsa replied:
Yeah, that makes sense and i will change it...

nunojsa commented:
Changes in V2:

- Fixed `special_probe()` coding style issue when added;
- Use `has_no_sample_clk` boolean to check when no sample clock is needed;
- Removed unneeded `owner` from `platform_driver` definition.
- Initialize `has_no_sample_clk` and `special_probe()` in some missing objects.

commodo commented on drivers/iio/adc/ad_adc.c:
```diff
@@ -208,12 +218,16 @@ static const struct iio_chan_spec adrv9009_obs_rx_channels[] = {
 };
 
 static const struct adc_chip_info adrv9009_obs_rx_single_chip_info = {
+	.special_probe = NULL,
```
just as a note: you can choose not to add `.special_probe = NULL,` as it is implied that it is NULL, if unspecified;
also fine to leave it

commodo replied:
same for `.has_no_sample_clk = false,` ; can be left out
